### PR TITLE
[OL-16] - added third bulk formula for catching errors and updated readme

### PR DESCRIPTION
**Pull request by kiawnna**
also replaced the microsoft dynamics transformation with one from sugar crm

### CODE REVIEW

**Comment by kiawnna** [comment]
@nick-bair @RickyAmparoCloud can't add you as a reviewer.

File: HowToUseBulkWithinAFormula/README.md [hunk 1]
```diff
@@ -51,25 +57,33 @@ You have two options for importing your resources. You can either import using t
...
-    * Make sure you remember which element you chose for CRM source. You will need to choose the same one in the next formula instance you create for Bulk Step 1.
+    * If using the example VDR for **Example 1**, you will need to authenticate an Element instance of both `Salesforce` and `Sugar CRM`. You will not see any transformations loaded from your VDR until you have authenticated instances. You will use one instance as a source Element and one as a destination (target) Element when creating your Formula instance below.
+    * **NOTE:** If you choose `Sugar CRM` as the source Element and `Salesforce` as the target, it's important to know that `Salesforce` does not allow for duplicate records to be created. Attempting to bulk upload records from `Sugar CRM` that already exist in `Salesforce` will result in an error for those records. If you'd like to test this direction and need new records created in your sandbox, consider checking out our example [SeedScript Repo](https://github.com/CloudElementsOpenLabs/scripts/tree/master/AddRecordsSeedScript), found in our OpenLabs [Scripts Repo](https://github.com/CloudElementsOpenLabs/scripts).
```
Comment (nick-bair) [inline HowToUseBulkWithinAFormula/README.md:63]: @kiawnna love this!

File: HowToUseBulkWithinAFormula/BulkStep2.json [hunk 2]
```diff
@@ -1,75 +1,105 @@
 {
-  "name": "Bulk Step 2",
+  "id": 41018,
+  "name": "BulkStep2",
```
Comment (nick-bair) [inline HowToUseBulkWithinAFormula/BulkStep2.json:3]: @kiawnna should we remove 'Act-On' from the description?

File: HowToUseBulkWithinAFormula/README.md [hunk 3]
```diff
@@ -1,30 +1,32 @@
-# Using Bulk in your formulas
-Working with bulk a formula can be challenging. Bulk by nature is an asynchronous job, while Formulas are a synchronous procedure. The trick to getting them to work together is webhooks.
+# Using Bulk in your Formulas
+Working with bulk a Formula can be challenging. Bulk by nature is an asynchronous job, while Formulas are a synchronous procedure. The trick to getting them to work together is webhooks.
```
Comment (nick-bair) [inline HowToUseBulkWithinAFormula/README.md:2]: @kiawnna  "The trick to getting them to work together is webhooks." should be "The trick to getting them to work together is bulk job callbacks" or something.  Calling it webhooks might confuse customers who associate that with event-driven formulas.

File: HowToUseBulkWithinAFormula/README.md [hunk 4]
```diff
@@ -1,30 +1,32 @@
...
 
-In this example, we will use two formulas--one creates the bulk job, specifying who to notify when the job completes, and one which recieves the 'job done' notification and consumes the results.  This configuration leverages our `Elements-Async-Callback-Url` header.
+In this example, we will use three Formulas--one creates the bulk job, specifying who to notify when the job completes, one which recieves the 'job done' notification and consumes the results, and lastly one which will catch any upload errors and notify you in they occur. This configuration leverages our `Elements-Async-Callback-Url` header.
```
Comment (nick-bair) [inline HowToUseBulkWithinAFormula/README.md:4]: @kiawnna update to 'and notify you in the case that they occur'

File: HowToUseBulkWithinAFormula/README.md [hunk 5]
```diff
@@ -51,25 +57,33 @@ You have two options for importing your resources. You can either import using t
...
     * For `object` under `Values`, enter: `MyContact` (or the name of your VDR).
-    * Capture the ID of the created formula instance. You will need to give the Bulk Step 1 formula the instance ID for the second formula. This will let the webhook trigger the correct formula when the bulk job is completed. 
-3. Create a formula instance for Bulk Step 1, using the same element for CRM source as you chose before. Under `Values`:
+    * For `thirdFormulaInstanceId`, enter the ID you captured from Step 2 when creating the Bulk Step 3 Formula.
+    * Capture the ID of newly the created Formula instance for Bulk Step 2. You will need to give the Bulk Step 1 Formula the instance ID for the second Formula. This will let the webhook trigger the correct Formula when the bulk job is completed. 
+4. Create a Formula instance for Bulk Step 1, using the same Element for CRM source as you chose before. Under `Values`:
     * For `object`, enter `MyContact` (or the name of your VDR).
     * Enter a cron job schedule (`0 0/10 0 ? * * *` will poll every 10 minutes).
```
Comment (nick-bair) [inline HowToUseBulkWithinAFormula/README.md:75]: @kiawnna Should we change this to a manual trigger?  My concern with a 10 minute cron is customers forget to turn it off.

File: HowToUseBulkWithinAFormula/TheDoctor/Formulas/BulkStep3.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    "active": true,
+    "singleThreaded": false,
+    "debugLoggingEnabled": true,
+    "configuration": [
+      {
+        "id": 82414,
+        "key": "email",
+        "name": "email",
+        "type": "value",
+        "description": "Enter an email to be notified in case of Bulk Upload failure.",
```
Comment (nick-bair) [inline HowToUseBulkWithinAFormula/TheDoctor/Formulas/BulkStep3.json:44]: @kiawnna Is there a step missing for the email notification?  Or are you using the failed execution notification?

**Review by nick-bair — CHANGES_REQUESTED** [review]
Thanks @kiawnna couple of comments below

File: HowToUseBulkWithinAFormula/README.md (hunk 1 above)
Comment (kiawnna) [reply HowToUseBulkWithinAFormula/README.md:63]: whoot whoot!

File: HowToUseBulkWithinAFormula/BulkStep2.json (hunk 2 above)
Comment (kiawnna) [reply HowToUseBulkWithinAFormula/BulkStep2.json:3]: yup! done!

File: HowToUseBulkWithinAFormula/README.md (hunk 3 above)
Comment (kiawnna) [reply HowToUseBulkWithinAFormula/README.md:2]: done!

File: HowToUseBulkWithinAFormula/README.md (hunk 4 above)
Comment (kiawnna) [reply HowToUseBulkWithinAFormula/README.md:4]: done!

File: HowToUseBulkWithinAFormula/README.md (hunk 5 above)
Comment (kiawnna) [reply HowToUseBulkWithinAFormula/README.md:75]: I agree. Changed to manual trigger and update readme.

File: HowToUseBulkWithinAFormula/TheDoctor/Formulas/BulkStep3.json (hunk 6 above)
Comment (kiawnna) [reply HowToUseBulkWithinAFormula/TheDoctor/Formulas/BulkStep3.json:44]: odd. I thought I had handled it all in that step. Updated so check it out now!

**Review by nick-bair — APPROVED** [review]
@kiawnna LGTM!

**Pull request merged**